test/core/iomgr/pollset_windows_starvation_test.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ *
+ * Copyright 2015 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/lib/iomgr/port.h"
+
+#include "src/core/lib/iomgr/tcp_server.h"
```

Comment:
Do you need tcp_server.h, and all the other header files? I do not check them closely, but please remove all the unnecessary headers. Thanks!

Also, I am not sure whether you create this test from scratch, or base it on some other grpc tests. ev_epollex_linux_test.cc may be a good reference, which tests a polling engine on Linux. To avoid running this test on platforms other than Windows, you can use the same trick as https://github.com/grpc/grpc/blob/master/test/core/iomgr/ev_epollex_linux_test.cc#L21. Basically you can guard the test under GRPC_WINSOCK_SOCKET.

Reply:
Removed (hopefully) all superfluous `#include`s.

I based this on something, but given I couldn't find `defined(WINDOWS)` equivalent, I ended up removing the `#if defined(..) ... #else .. #endif` blocks, hoping specifyinig the test as Windows-only would not try compiling it on other platforms. Guess I was wrong. :)